### PR TITLE
Reversing a string in java

### LINKED ISSUE #98: Reversing a string using stacks
for example-prateek
output-keetarp

### PR DESCRIPTION
**Pull request by prateek-jpg**
Reversing a string using stacks.Please consider this under GSSOC'21.
Fixes #98

### CODE REVIEW

**Review by raksha009 — CHANGES_REQUESTED** [review]
Plz remove the extra file

**Comment by prateek-jpg** [comment]
@raksha009  Extra files removed.

File: Stack/Reverse_String.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+public class Reverse {
+
+	public static void main(String[] args) {
+		
+	Scanner sc=new Scanner(System.in);
+		
+        System.out.print("Enter a String:");
+        
+	String s=sc.nextLine();												//Enter the string you want to reverse
```
Comment (rudrakshi99) [inline Stack/Reverse_String.java:19]: Do not add too many spaces at the starting of comments. Add the comments just above the code.

File: Stack/Reverse_String.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        int l=s.length();												//Storing the length of the string
```
Comment (rudrakshi99) [inline Stack/Reverse_String.java:21]: Do not add too many spaces at the starting of comments. Add the comments just above the code.

File: Stack/Reverse_String.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        Stack<Character> a=new Stack<>();             									//Creating a stack named "a" 
+
+       
+
+        for(char ch:s.toCharArray()){			
+																											//iterating through the string using forEach loop
+            a.push(ch);													//adding each character to the stack
```
Comment (rudrakshi99) [inline Stack/Reverse_String.java:29]: Do not add too many spaces at the starting of comments. Add the comments just above the code.

File: Stack/Reverse_String.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        
+	}
+
+        s="";
+
+        while(l-->0){
+					
+            s=s+a.pop();												//as pop occurs from top therefore while popping a character we add it to a string which is stored in reverse order
```
Comment (rudrakshi99) [inline Stack/Reverse_String.java:37]: Do not add too many spaces at the starting of comments. Add the comments just above the code.

File: Stack/Reverse_String.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		
+	Scanner sc=new Scanner(System.in);
+		
+        System.out.print("Enter a String:");
+        
+	String s=sc.nextLine();												//Enter the string you want to reverse
+
+        int l=s.length();												//Storing the length of the string
+
+        Stack<Character> a=new Stack<>();             									//Creating a stack named "a"
```
Comment (rudrakshi99) [inline Stack/Reverse_String.java:23]: Do not add too many spaces at the starting of comments. Add the comments just above the code.

**Review by rudrakshi99 — CHANGES_REQUESTED** [review]

File: Stack/Reverse_String.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+/* Output
+1.Enter a String:Prateek
+   Output-keetarP
+2.Enter a String:This is the end of world
+	Output-dlrow fo dne eht si sihT
+	
+	
+	
+	Time Complexity-O(n)
```
Comment (rudrakshi99) [inline Stack/Reverse_String.java:57]: use `:` instead of `-`.

File: Stack/Reverse_String.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+	Space Complexity-O(n)
```
Comment (rudrakshi99) [inline Stack/Reverse_String.java:58]: use `:` instead of  `-`.

**Comment by prateek-jpg** [comment]
@rudrakshi99 I have removed extra spaces which were unnecessary and made the changes mentioned. Kindly check it and let me know if any other change is required.

File: Stack/Reverse_String.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+        s="";
+
+        while(l-->0){
+	//as pop occurs from top therefore while popping a character we add it to a string which is stored in reverse order			
+            s=s+a.pop(); 					
+        }
+		
+        System.out.println("Output-"+s);           
+
```
Comment (raksha009) [inline Stack/Reverse_String.java:39]: Remove extra space

File: Stack/Reverse_String.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
```
Comment (raksha009) [inline Stack/Reverse_String.java:41]: Remove extra space

**Review by raksha009 — CHANGES_REQUESTED** [review]
Link it to the folder.

**Comment by prateek-jpg** [comment]
@raksha009  Removed extra spaces and also linked the file in readme.

**Review by raksha009 — APPROVED** [review]

**Comment by prateek-jpg** [comment]
Hiii @rudrakshi99 
Please let me know if there are any other changes required in my PR.

**Comment by Manish5237** [comment]
Hi ,

I am GSSOC'21 participant.
Please assign me this. I will do this in C with proper comments and test cases.

@rudrakshi99

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
this should be linked to string folder, not stack

**Comment by prateek-jpg** [comment]
@tarun26091999  have linked to string folder.

**Comment by naqeeb2710** [comment]
can i work on this issue?

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
where is your code?

**Comment by rudrakshi99** [comment]
Please resolve merge conflicts.

**Pull request closed without merging**